crates/katana/storage/provider/src/lib.rs

```diff
@@ -379,3 +380,24 @@ where
         self.provider.block_env_at(id)
     }
 }
+
+impl<Db> MessagingProvider for BlockchainProvider<Db>
+where
+    Db: MessagingProvider,
+{
+    fn get_send_from_block(&self) -> ProviderResult<Option<BlockNumber>> {
+        self.provider.get_send_from_block()
+    }
+
+    fn set_send_from_block(&self, send_from_block: BlockNumber) -> ProviderResult<()> {
+        self.provider.set_send_from_block(send_from_block)
+    }
+
+    fn get_gather_from_block(&self) -> ProviderResult<Option<BlockNumber>> {
+        self.provider.get_gather_from_block()
+    }
+
+    fn set_gather_from_block(&self, gather_from_block: BlockNumber) -> ProviderResult<()> {
+        self.provider.set_gather_from_block(gather_from_block)
+    }
+}
```

Comment:
As seen, we should also keep track of a pointer on the message.
L1->L2, we can use the nonce of the message.
L2->L1, let's use the index of the message in the block as they are always ordered.